src/QsCompiler/Compiler/Templates/CppInterop.cs

```diff
@@ -0,0 +1,151 @@
+                DataType.RangeType => "RangeTuple",
+                DataType.ResultType => "char",
+                DataType.StringType => "string",
+                DataType.ArrayType => arg.ArrayType switch
+                {
+                    DataType.BoolType => "vector<char>",
+                    DataType.IntegerType => "vector<int64_t>",
+                    DataType.DoubleType => "vector<double_t>",
+                    DataType.PauliType => "std::vector<PauliId>",
+                    DataType.RangeType => "vector<RangeTuple>",
+                    DataType.ResultType => "vector<char>",
+                    DataType.StringType => "vector<string>",
+                },
+            };
+        }
+
+        public static List<Argument> GetSortedArguments(EntryPointOperation op)
+        {
+            op.Arguments.Sort((a, b) => a.Position.CompareTo(b.Position));
+            return op.Arguments;
+        }
+
+        public static string? CppVarInitialValue(Argument arg)
+        {
+            return arg.Type switch
+            {
+                DataType.BoolType => "InteropFalseAsChar",
+                DataType.IntegerType => "0",
+                DataType.DoubleType => "0.0",
+                DataType.PauliType => "PauliId::PauliId_I",
+                DataType.RangeType => null,
+                DataType.ResultType => "InteropResultZeroAsChar",
+                DataType.StringType => null,
+                DataType.ArrayType => arg.ArrayType switch
+                {
+                    DataType.BoolType => null,
+                    DataType.IntegerType => null,
+                    DataType.DoubleType => null,
+                    DataType.PauliType => null,
+                    DataType.RangeType => null,
+                    DataType.ResultType => null,
+                    DataType.StringType => null,
+                },
+            };
+        }
+
+        public static string? TransformationType(Argument arg)
+        {
+            return arg.Type switch
+            {
+                DataType.BoolType => "BoolAsCharMap",
+                DataType.IntegerType => null,
+                DataType.DoubleType => null,
+                DataType.PauliType => "PauliMap",
+                DataType.RangeType => null,
+                DataType.ResultType => "ResultAsCharMap",
+                DataType.StringType => null,
+                DataType.ArrayType => arg.ArrayType switch
+                {
+                    DataType.BoolType => "BoolAsCharMap",
+                    DataType.IntegerType => null,
+                    DataType.DoubleType => null,
+                    DataType.PauliType => "PauliMap",
+                    DataType.RangeType => null,
+                    DataType.ResultType => "ResultAsCharMap",
+                    DataType.StringType => null,
+                },
+            };
+        }
+
+        public static bool AtLeastOneArrayArgument(EntryPointOperation op)
```

Comment:
To make this more generic, this could receive a `DataType` and returns `true` if at least one argument is of that type.

Additionally, since it will be more generic a good name for this method would be `ContainsArgumentType`.

Reply:
done